src/meta/meta_ckpt.c

```diff
@@ -510,6 +513,17 @@ __wt_meta_ckptlist_get(
         ckpt = &ckptbase[slot];
         ckpt->order = (slot == 0) ? 1 : ckptbase[slot - 1].order + 1;
         __wt_seconds(session, &ckpt->sec);
+        /*
+         * Update time value for most recent checkpoint, not letting it move backwards. It is
+         * possible to race here, so use atomic CAS. This code relies on the fact that anyone we
+         * race with will only increase (never decrease) the most recent checkpoint time value.
+         */
+        while (true) {
```

Comment:
You had no way of knowing this. Our coding style only uses `for (;;)` for forever loops.

Reply:
The style guide on the wiki covers a lot of ground with, "If in doubt find an existing example in the source code and copy it." :-)